tests/unit/disco/components/TestAddon.js

```diff
@@ -491,112 +610,74 @@ describe(__filename, () => {
   });
 
   describe('addon with type lightweight theme', () => {
-    const renderWithLightweightTheme = (props = {}) => {
-      const addon = {
-        ...result,
-        type: ADDON_TYPE_THEME,
-        previews: [],
-      };
-
-      return renderAddon({ addon, ...props });
+    const renderLightweightTheme = ({ addonProps, ...otherProps } = {}) => {
+      return render({
+        addonProps: {
+          type: ADDON_TYPE_THEME,
+          ...addonProps,
+        },
+        ...otherProps,
+      });
     };
 
     it('renders a ThemeImage', () => {
-      const root = renderWithLightweightTheme();
+      const root = renderLightweightTheme();
 
       expect(root.find(ThemeImage)).toHaveLength(1);
     });
 
     it('makes the ThemeImage clickable when add-on manager is available', () => {
-      const addon = {
-        ...result,
+      const addonProps = {
+        id: 'some-theme-id',
+        name: 'some-theme-name',
         type: ADDON_TYPE_THEME,
-        previews: [],
       };
+
+      store.dispatch(
+        setInstallState({
+          // We use this function because the GUID is changed for lightweight
+          // themes (to mimic Firefox's internal behavior).
```

Comment:
Does the magic guid swapping still work? I am not set up very well to test it locally.

Reply:
Yep, it works because `createInternalAddon()` calls the `getGuid()` function.